File: pgm/ABC_Java/ABC_255_A.java
import java.io.*;
// import java.math.*;
// import java.util.*;
// import java.util.Map.*;
// import java.util.stream.*;

public class Main {
    public static void main (String[] args) throws Exception {
        BufferedReader reader =
            new BufferedReader (new InputStreamReader (System.in));
        final String[] arr = reader.readLine().split (" ");
        final int zero = 0;
        final int one = 1;
        final int two = 2;
        final int R = Integer.parseInt (arr[0]);
        final int C = Integer.parseInt (arr[1]);
        final int[][] A = new int[two][two];
        for (int i = 0; i < two; i++) {
            String[] row = reader.readLine().split (" ");
            A[i][zero] = Integer.parseInt (row[zero]);
            A[i][one] = Integer.parseInt (row[one]);
        }
        System.out.println (A[R - 1][C - 1]);
        return;
    }
}

File: pgm/ABC_Kotlin/ABC_255_A.kt
// import java.io.BufferedOutputStream
// import java.io.BufferedReader
// import java.io.DataInputStream
// import java.io.FileInputStream
// import java.io.IOException
// import java.io.InputStreamReader
// import java.io.PrintWriter
// import java.lang.StringBuilder
// import java.time.LocalTime
// import java.time.format.DateTimeFormatter
// import java.util.Scanner
// import java.util.StringTokenizer

// import kotlin.math.*

const val space: String = " "
const val one: Int = 1

fun main() {
    val (R: Int, C: Int) = readln().split(space).map { it.toInt() }
    val A: Array<IntArray> = arrayOf(
        readln().split(space).map { it: String -> it.toInt() }.toIntArray(),
        readln().split(space).map { it: String -> it.toInt() }.toIntArray()
    )
    println(A[R - one][C - one])
    kotlin.system.exitProcess(0)
}